### DISCUSSION TITLE
Improve inference of parameter types in function decorated with `pytest.mark.parametrize`

### DISCUSSION THREAD

**Discussion by Sachaa-Thanasius** [microsoft/pylance-release#6024] [Enhancement]
Pylance can currently infer the types of parameters in a function decorated with `pytest.mark.parametrize`, but only if the first argument of `parametrize` is a single string instead of a list or a tuple. I think it would be worth extending the analysis of the first argument to lists and tuples as well, as I think they're pretty common patterns for defining the list of parameters as well.

**Pylance info:**
Version: `v2024.6.100 (pre-release)`
Settings:
(can include more if need be, I just checked the pytest-related ones)
```json
"python.analysis.enablePytestSupport": true
"python.analysis.inlayHints.pytestParameters": true
```

**Screenshot displaying lack of inference:**

![image](https://github.com/microsoft/pylance-release/assets/111999343/b2df9b28-9386-483a-9910-7126b1ece630)

**Reproduction code:**
```py
# tests/test_thing.py
import pytest

@pytest.mark.parametrize(
    "first, second",
    [
        ("1", 1),
        ("2", 2),
    ],
)
def test_function_string(first, second):
    assert int(first) == second


@pytest.mark.parametrize(
    ["first", "second"],
    [
        ("1", 1),
        ("2", 2),
    ],
)
def test_function_list(first, second):
    assert int(first) == second


@pytest.mark.parametrize(
    ("first", "second"),
    [
        ("1", 1),
        ("2", 2),
    ],
)
def test_function_tuple(first, second):
    assert int(first) == second
```

**Reproduction folder structure:**
```
.
├── tests
│   ├── __init__.py
│   └── test_thing.py
```

**Comment by heejaechang** [comment]
let's see how much upvote we get.
